server/server.go

```diff
@@ -835,14 +836,31 @@ func (server *Server) loadEntryPointConfig(entryPointName string, entryPoint *En
 	return rewrite, nil
 }
 
-func (server *Server) buildDefaultHTTPRouter() *mux.Router {
+func (server *Server) buildDefaultHTTPRouter(noRouteResponseCode int) *mux.Router {
 	router := mux.NewRouter()
-	router.NotFoundHandler = http.HandlerFunc(notFoundHandler)
+	router.NotFoundHandler = newNotFounderHandler(noRouteResponseCode)
 	router.StrictSlash(true)
 	router.SkipClean(true)
 	return router
 }
 
+func newNotFounderHandler(responseCode int) http.Handler {
+	if responseCode == 0 {
+		responseCode = http.StatusNotFound
+	}
+	if http.StatusText(responseCode) == "" {
+		log.Warnf("Wrong NoRouteResponseCode %d configured. Falling back to 404 (http.StatusNotFound)", responseCode)
```

Comment:
@timoreimann indeed your comment addressed it. During the course I changed implementation now a bit, pushing the configuration validation and fallback, as well as the logging part, to the `buildDefaultHTTPRouter` function. This way we don't need errors. WDYT?